src/poetry/core/semver/version_range.py

```diff
@@ -69,6 +69,11 @@ def is_simple(self) -> bool:
 
     def allows(self, other: Version) -> bool:
         if self._min is not None:
+            if self._min.is_local() and (
+                not other.is_local() or self._min.local != other.local
```

Comment:
That's not completely correct. It actually has to be something like  `self._min.local < other.local`.

Reply:
IIRC, local labels are to be treated as strings, so they should not be ordered when comparing.

Reply:
This means `>=1+local.1` includes `1+local.1` and `2+local.1` but not `1+local.2`.

Reply:
Yeah, it might not be suitable for stuff like `1.10.0+cu113` and `1.10.0+cpu` but that's what PEP440 says.